src/components/Dashboard/APIcall.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable */
+import React, { useState } from 'react'
+import GitHubLogin from 'react-github-login'
+import { Button } from '@aragon/ui'
+import './test.css'
+import { Octokit } from '@octokit/rest'
+
+function APIcall() {
+  const [boolButtonText, setBoolButtonText] = useState('CONNECT')
+
+  const [boolSwitch, setboolSwitch] = useState(false)
+
+  const onSuccess = function(details) {
+    const code = details.code
+
+    fetch(`https://gitfund-oauth.herokuapp.com/authenticate/${code}`)
+      .then((response) => response.json())
+      .then(({ token }) => localStorage.setItem('ACCESS TOKEN', token))
+
+    if (boolButtonText === 'CONNECT') {
```

Comment:
I think this logic just toggles the button text. Make sure onSuccess it is always "DISCONNECT" and you need to add a condition in Button so that GitHubLogin is invoked only when the text is "CONNECT" and delete access token when text is "DISCONNECT". I thought that this was implemented.

Reply:
Currently when I refresh page or switch between the tabs it displays "CONNECT" text again. Make sure that the status persists.

Reply:
But even If we refresh the page, the wallet has to be persistent too right?

Reply:
Ideally yes, but the Aragon wallet doesn't perform that. We will fix it soon. For now, let's make the GitHub connection persistent through local storage.

Reply:
Solved the persistent problem